apps/client/src/modules/platform/pipelineruns/utils/statusIcons.ts

```diff
@@ -0,0 +1,69 @@
+import { STATUS_COLOR } from "@/k8s/constants/colors";
+import { TektonResultStatus, DecodedPipelineRunCondition } from "@my-project/shared";
+import { CircleCheck, CircleX, Clock, Loader2, ShieldQuestion, LucideIcon } from "lucide-react";
+
+export interface StatusIconConfig {
+  Icon: LucideIcon;
+  color: string;
+  title: string;
+  isSpinning?: boolean;
+}
+
+/**
+ * Get status icon configuration for Tekton Result status
+ * Used in list/table views showing historical pipeline run results
+ */
+export const getTektonResultStatusIcon = (status: TektonResultStatus | undefined): StatusIconConfig => {
+  switch (status) {
```

Comment:
instaed of switch case create an object mapper

Reply:
done